standard/mock/mock_team_template.lua

```diff
@@ -0,0 +1,256 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Mock/TeamTemplate
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+-- Warning: This mock ignores the posibility of image/imagedark being unset/empty
+-- and never uses legacyimage/legacyimagedark for that reason
+
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local MockData = Lua.import('test_assets.team_template_data')
+
+local NOT_FOUND_ERROR = '<div class="error">No team template exists for name "${name}".</div>'
+local MISSING_INPUT_ERROR = 'bad argument #1 to "name" (string or number expected, got nil)'
+
+local mockTeamTemplate = {}
+
+-- temp copy the orig functions so we can retrieve them later again when tearing down the mock
+local _teamTemplate = Table.copy(mw.ext.TeamTemplate)
+
+function mockTeamTemplate.setUp()
+	mw.ext.TeamTemplate.raw = mockTeamTemplate.raw
+	mw.ext.TeamTemplate.raw_historical = mockTeamTemplate.raw_historical
+	mw.ext.TeamTemplate.teamexists = mockTeamTemplate.teamexists
+	mw.ext.TeamTemplate.team = mockTeamTemplate.team
+	mw.ext.TeamTemplate.team2 = mockTeamTemplate.team2
+	mw.ext.TeamTemplate.teamshort = mockTeamTemplate.teamshort
+	mw.ext.TeamTemplate.team2short = mockTeamTemplate.team2short
+	mw.ext.TeamTemplate.teambracket = mockTeamTemplate.teambracket
+	mw.ext.TeamTemplate.teamicon = mockTeamTemplate.teamicon
+	mw.ext.TeamTemplate.teamimage = mockTeamTemplate.teamimage
+	mw.ext.TeamTemplate.teampage = mockTeamTemplate.teampage
+	mw.ext.TeamTemplate.teampart = mockTeamTemplate.teampart
+	mw.ext.TeamTemplate.raw = mockTeamTemplate.raw
+end
+
+function mockTeamTemplate.tearDown()
+	mw.ext.TeamTemplate = _teamTemplate
+end
+
+---@param teamtemplate string
+---@param date string|number?
+---@return teamTemplateData?
+function mockTeamTemplate.raw(teamtemplate, date)
```

Comment:
Doesn't treat `historicaltemplate` correctly, but we can leave it out

Reply:
https://discord.com/channels/93055209017729024/443856413052239882/873097457624547348

This is how `historicaltemplate` field work

Reply:
ah so remove it if the sub template is called directly

Reply:
Or add it if the historicaltemplate is called, I'd say

Reply:
then i would have to rework the alias resolving :P